### PR TITLE
fix: avoid conflicts when using multiple roots on the same page

### PR DESCRIPTION
**Pull request by LiKang6688**
save reactRoot as a state 
create a new reactRoot when there is a new element to render on

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -76,8 +81,9 @@ export default function supernova(env) {
         if (layout && tableData && !env.carbon) {
           registerLocale(translator);
           const changeSortOrder = sortingFactory(model);
-          render(rootElement, {
+          render({
             rootElement,
+            reactRoot,
```
Comment (haxxmaxx) [inline src/index.js:86]: Don't put it in this object, no need to pass it to the rest of the app

File: src/index.js [hunk 2]
```diff
@@ -25,8 +27,12 @@ import { mount, render, teardown } from './table/Root';
 import muiSetup from './mui-setup';
 import tableThemeColors from './table-theme-colors';
 
+const initialPageInfo = () => ({
```
Comment (haxxmaxx) [inline src/index.js:30]: This could just be an object

File: src/table/Root.jsx [hunk 3]
```diff
@@ -1,19 +1,16 @@
-import { createRoot } from 'react-dom/client';
 import React from 'react';
 import { StyleSheetManager } from 'styled-components';
 import { ThemeProvider } from '@mui/material/styles';
 import rtlPluginSc from 'stylis-plugin-rtl-sc';
 import TableWrapper from './components/TableWrapper';
 
-let reactRoot;
+let root;
```
Comment (haxxmaxx) [inline src/table/Root.jsx:7]: Do we need this? Can't we pass it to teardown as well? Might as well avoid global variables if we can

File: src/index.js (hunk 1 above)
Comment (LiKang6688) [reply src/index.js]: Fixed

File: src/index.js (hunk 2 above)
Comment (LiKang6688) [reply src/index.js]: Sure. Done

File: src/table/Root.jsx (hunk 3 above)
Comment (LiKang6688) [reply src/table/Root.jsx]: No, we do not. Good idea. Done

File: src/index.js [hunk 4]
```diff
@@ -95,6 +99,7 @@ export default function supernova(env) {
           });
         }
       }, [
+        rootElement,
```
Comment (haxxmaxx) [inline src/index.js:102]: If anything, this should have the reactRoot as dependency, otherwise we could still be passing the old reactRoot

Comment (LiKang6688) [reply src/index.js:102]: good find. I mistakenly remove the line.

Comment (haxxmaxx) [reply src/index.js:102]: you wont need rootElement though since if `rootElement` changes `reactRoot` will change too

Comment (LiKang6688) [reply src/index.js]: that's true. Updated.

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**